src/vulkan/pipeline.h

```diff
@@ -83,29 +83,34 @@ class Pipeline {
   VkPipeline pipeline_ = VK_NULL_HANDLE;
   VkPipelineLayout pipeline_layout_ = VK_NULL_HANDLE;
 
-  std::vector<VkDescriptorSetLayout> descriptor_set_layouts_;
-  std::vector<VkDescriptorPool> descriptor_pools_;
-  std::vector<VkDescriptorSet> descriptor_sets_;
-
   VkDevice device_ = VK_NULL_HANDLE;
   VkPhysicalDeviceMemoryProperties memory_properties_;
   std::unique_ptr<CommandBuffer> command_;
 
  private:
+  struct DescriptorSet {
```

Comment:
Should this just be part of the Descriptor object? It seems like they're always used together?

Reply:
Hmm .. I don't think so.

- `DescriptorSet` is not 1:1 matched with `Descriptor`. `DescriptorSet` has one or more `Descriptor`.
- `Descriptor` does not use `DescriptorSet` inside. It is only used by `Pipeline`.

Reply:
So, how are they related? Is a DescriptorSet required to make a Descriptor? One DescriptorSet can make multiple Descriptors but the DescriptorSets have to be contiguous? Can we change the DescriptorSets struct to have a vector<Descriptor> in it which is all the descriptors for a given DescriptorSet?